src/ddmd/s2ir.d

```diff
@@ -691,19 +691,16 @@ extern (C++) class S2irVisitor : Visitor
     {
         Blockx *blx = irs.blx;
 
-        //printf("SwitchErrorStatement.toIR()\n");
-        elem *e;
-        if (global.params.checkAction == CHECKACTION.C)
+        if (s.exp)
         {
-            e = callCAssert(irs, s.loc, null, null, "no switch default");
+            //printf("SwitchErrorStatement.toIR(), exp = %s\n", s.exp ? s.exp.toChars() : "");
+            incUsage(irs, s.loc);
+            block_appendexp(blx.curblock, toElemDtor(s.exp, irs));
         }
         else
         {
-            auto efilename = el_ptr(toSymbol(cast(Dsymbol)blx._module));
-            auto elinnum = el_long(TYint, s.loc.linnum);
-            e = el_bin(OPcall, TYvoid, el_var(getRtlsym(RTLSYM_DSWITCHERR)), el_param(elinnum, efilename));
+            assert(0);
```

Comment:
It's a good style to provide an informative error message or do sth. like 

```d
assert(s.exp !is null, "SwitchErrorStatement needs to have a valid Expression.") 
```

and skip the `if` entirely.

Reply:
> It's a good style to provide an informative error message or do sth. like...

@WalterBright disagrees: https://github.com/dlang/dmd/pull/7079#discussion_r132904181

> Some people like assert messages, I don't. They bloat up the compiler. Leaving the message as a comment in the source code is plenty good enough.

I really don't mind either way, just an FYI.